contracts/Plasma.sol

```diff
@@ -153,55 +155,57 @@ contract Plasma is Ownable {
 
   // challenge
   function challengeWithdrawDeposit(uint _blockIndex, bytes _transactionBytes, bytes _proof, bytes signature) public returns (bool success) {
-    Block memory block = blocks[_blockIndex];
+    Block memory newBlock = blocks[_blockIndex];
     Transaction.TX memory transaction = Transaction.createTransaction(_transactionBytes);
 
-    require(deposits[transaction.uid].amount > 0); // check if the deposit exists
+    require(deposits[transaction.uid].amount > 0, "deposits[transaction.uid].amount > 0"); // check if the deposit exists
 
     //require(exits[transaction.uid].timestamp >= block.timestamp); // check if challenge timeout is off
 
     bytes32 hash = Transaction.hashTransaction(transaction);
 
-    require(MerkleProof.verifyProof(_proof, block.merkleRoot, hash, transaction.uid)); // check is TX exists
+    // check is TX exists
+    require(MerkleProof.verifyProof(_proof, newBlock.merkleRoot, hash, transaction.uid), "MerkleProof.verifyProof(...)");
 
     address exitor = exits[transaction.uid].exitor;
 
-    require(Transaction.checkSig(exitor, hash, signature)); // check if the correspondent signature correct
+    // check if the correspondent signature correct
+    require(Transaction.checkSig(exitor, hash, signature), "Transaction.checkSig(exitor, hash, signature)");
 
-    require(exitor != transaction.newOwner); // check if the owner has been changed
+    require(exitor != transaction.newOwner, "exitor != transaction.newOwner"); // check if the owner has been changed
 
     delete exits[transaction.uid]; // delete exit
     // TODO: prevent further challengeWithdrawDeposit calls with the same UID
 
-    DepositWithdrawChallenged(msg.sender, exitor, transaction.uid, _blockIndex);
+    emit DepositWithdrawChallenged(msg.sender, exitor, transaction.uid, _blockIndex);
 
     return true;
   }
 
   // temp methods
-  function verifyProof(bytes _proof, bytes32 _root, bytes32 _leaf, uint256 _index) public constant returns (bool success) {
+  function verifyProof(bytes _proof, bytes32 _root, bytes32 _leaf, uint256 _index) public pure returns (bool success) {
     return MerkleProof.verifyProof(_proof, _root, _leaf, _index);
   }
 
-  function proveTX(uint _blockIndex, bytes _transactionBytes, bytes _proof) public constant returns (bool success) {
-    Block memory block = blocks[_blockIndex];
+  function proveTX(uint _blockIndex, bytes _transactionBytes, bytes _proof) public view returns (bool success) {
+    Block memory newBlock = blocks[_blockIndex];
```

Comment:
why 'new'?

Reply:
because `block` is global variable in solidity and compiler give us a warning for it.